### DISCUSSION TITLE
Where is lcd_periph_signals defined

### DISCUSSION THREAD

**Discussion by RonKsr** [lovyan03/LovyanGFX#632] [Q&A] [answered]
LovyanGFX version 1.1.16, file Bus_RGB.cpp.
Included the following:
```
#include <LovyanGFX.hpp>
#include <lgfx/v1/platforms/esp32s3/Panel_RGB.hpp>
#include <lgfx/v1/platforms/esp32s3/Bus_RGB.hpp>

```
 Visual Studio issues an error on line 159 and 307
```
Bus_RGB.cpp: In member function virtual bool lgfx::v1::Bus_RGB::init()
 
Bus_RGB.cpp: 159:19: error: 'lcd_periph_signals' was not declared in this scope; did you mean 'lcd_periph_rgb_signals'?
   159 |       auto sigs = lcd_periph_signals.panels[_cfg.port]
   |                   ^~~~~~~~~~~~~~~~~~
   |                   lcd_periph_rgb_signals
 
Bus_RGB.cpp: 307:31: error: 'lcd_periph_signals' was not declared in this scope; did you mean 'lcd_periph_rgb_signals'?
   307 |     esp_intr_alloc_intrstatus(lcd_periph_signals.panels[_cfg.port].irq_id, isr_flags
   |                               ^~~~~~~~~~~~~~~~~~
   |                               lcd_periph_rgb_signals

```
My question is how or where is` lcd_periph_signals` defined?
I don't understand where VS got `lcd_periph_rgb_signals` from because I cant find that either.

An help or insight apricated.

**Comment by tobozo** [comment]
Hi,

It's from an esp-idf component brought in by the `#include "soc/lcd_periph.h"` directive.

The error can be solved by using the latest version of espressif32 platform which comes with a more recent version of esp-idf.

If you're already using the latest official espressif platform and you know it won't solve the error, you can try using this [non official platformio core](https://github.com/pioarduino/platform-espressif32) instead.

**Reply by RonKsr** [reply, marked as answer]
ToBozo, thank you

I traced down my problem. You provided the clue with  `#include "soc/lcd_periph.h"` 
For some reason, the esp-idf component version 3.1.0-RC2 changed the
name from `lcd_periph_signals` to `lcd_preiph_rgb_signals` making
LovyanGFX version 1.1.16 throw the errors since it requires` lcd_periph_signals`
for Bus_RGB.cpp.

Changing the ESP version 3.1.0-RC2 to ESP version 3.1.0-RC1
solved the issue.

My thanks again.
